src/fsharp/service/FSharpCheckerResults.fs

```diff
@@ -1883,6 +1883,21 @@ type FSharpCheckFileResults
                             let symbol = FSharpSymbol.Create(cenv, symbolUse.Item)
                             yield FSharpSymbolUse(scope.TcGlobals, symbolUse.DisplayEnv, symbol, symbolUse.ItemOccurence, symbolUse.Range) |])
 
+    member __.GetAllUsesOfAllSymbolsInFileByPredicate(predicate: (FSharpSymbolUse -> bool), ?cancellationToken: CancellationToken ) = 
```

Comment:
It feels like `GetAllUsesOfAllSymbolsInFile` could just return a `seq { ... }` now (which can have cancellation built in through the IDisposable - we'd need a Seq.toArray at the end which took a cancellation token), with the filter at each callsite.

But I do appreciate this is quite perf sensitive so would need to be tested.

Reply:
Narrator: it's an improvement.